x-pack/plugin/esql/src/test/java/org/elasticsearch/xpack/esql/optimizer/LocalPhysicalPlanOptimizerTests.java

```diff
@@ -571,11 +737,126 @@ public void testQueryStringFunctionMultipleWhereClauses() {
+        VerificationException ve = expectThrows(
+            VerificationException.class,
+            () -> plannerOptimizer.plan("from test | where " + functionInvocation + " or length(first_name) > 12", IS_SV_STATS)
+        );
+        assertThat(
+            ve.getMessage(),
+            containsString(
+                LoggerMessageFormat.format(
+                    null,
+                    "1:19: Invalid condition [{} or length(first_name) > 12]. "
+                        + "Function {} can't be used as part of an or condition that includes [length(first_name) > 12]",
+                    functionInvocation,
+                    functionName
+                )
+            )
+        );
+        ve = expectThrows(
+            VerificationException.class,
+            () -> plannerOptimizer.plan(
+                "from test | where (" + functionInvocation + " and emp_no < 1000) or (length(first_name) > 12 and emp_no > 1000)",
+                IS_SV_STATS
+            )
+        );
+        assertThat(
+            ve.getMessage(),
+            containsString(
+                LoggerMessageFormat.format(
+                    null,
+                    "1:19: Invalid condition [({} and emp_no < 1000) or (length(first_name) > 12 and emp_no > 1000)]. "
+                        + "Function {} can't be used as part of an or condition that includes [length(first_name) > 12 and emp_no > 1000]",
+                    functionInvocation,
+                    functionName
+                )
+            )
+        );
+    }
+
+    public void testMatchFunctionDisjunctionNonPushableClauses() {
+        assumeTrue("skipping because MATCH function is not enabled", EsqlCapabilities.Cap.MATCH_FUNCTION.isEnabled());
+        String queryText = """
+            from test
+            | where match(first_name, "Peter") or length(last_name) > 10
+            """;
+
+        VerificationException ve = expectThrows(VerificationException.class, () -> plannerOptimizer.plan(queryText, IS_SV_STATS));
+        assertThat(
+            ve.getMessage(),
+            containsString(
+                "Invalid condition [match(first_name, \"Peter\") or length(last_name) > 10]. "
+                    + "Function MATCH can't be used as part of an or condition that includes [length(last_name) > 10]"
+            )
+        );
+    }
+
+    public void testMatchFunctionIsNotNullable() {
+        assumeTrue("skipping because MATCH function is not enabled", EsqlCapabilities.Cap.MATCH_FUNCTION.isEnabled());
+
+        String queryText = """
+            row n = null | eval text = n + 5 | where match(text::keyword, "Anna")
+            """;
+
+        VerificationException ve = expectThrows(VerificationException.class, () -> plannerOptimizer.plan(queryText, IS_SV_STATS));
+        assertThat(
+            ve.getMessage(),
+            containsString("[MATCH] cannot operate on [text::keyword], which is not a field from an index mapping")
+        );
+    }
+
```

Comment:
Move these tests catching `VerificationException` to `LogicalPlanOptimizerTests`.

Reply:
Done in 87807ed508002848138e2871620fe2a7c6ce61fc